language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionTypeDescriptorNodeContext.java

```diff
@@ -76,12 +81,37 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
             ruleContext = RuleContext.PARAMETER_CTX;
         } else if (this.withinReturnKWContext(context, node)) {
             completionItems.add(new SnippetCompletionItem(context, Snippet.KW_RETURNS.get()));
+        } else if (node.parent().kind() == SyntaxKind.OBJECT_FIELD) {
+            SnippetCompletionItem initFuncCompletionItem =
+                    new SnippetCompletionItem(context, Snippet.DEF_INIT_FUNCTION.get());
+            Token funcKW = node.functionKeyword();
+            int endOffset = getEndPosWithoutNewLine(funcKW);
+            Range range = PositionUtil.toRange(funcKW.textRange().startOffset(), endOffset,
+                    context.currentDocument().get().textDocument());
+            TextEdit textEdit = new TextEdit(range, "");
+            initFuncCompletionItem.getCompletionItem().setAdditionalTextEdits(List.of(textEdit));
+            completionItems.add(initFuncCompletionItem);
+            completionItems.add(new SnippetCompletionItem(context, Snippet.DEF_FUNCTION.get()));
         }
         this.sort(context, node, completionItems, ruleContext);
 
         return completionItems;
     }
 
+    private int getEndPosWithoutNewLine(Token token) {
+        int end = token.textRangeWithMinutiae().endOffset();
```

Comment:
Shall we add a comment specifying the scenario

Reply:
Added